client/src/components/MyProfile.js

```diff
@@ -1,9 +1,272 @@
-import React from 'react'
+import { React, useState, useEffect } from 'react';
+import '../styles/MyProfile.css';
+import Button from "@mui/material/Button";
+import EditDetailsModal from '../modals/EditDetailsModal';
+
+function EventItem(props)
+{
+
+  // These states store the original event data
+  const [Firstname, setFirstname] = useState(props.Firstname);
+  const [Lastname, setLastname] = useState(props.Lastname);
+  const [Phone, setPhone] = useState(props.Phone);
+  const [Address, setAddress] = useState(props.Address);
+  const [Postalcode, setPostalcode] = useState(props.Postalcode);
+  const [City, setCity] = useState(props.City);
+  const [Country, setCountry] = useState(props.Country);
+  const [Email, setEmail] = useState(props.Email);
+
+  // These states store the data that is edited
+  const [edit, setEdit] = useState(false);
+  const [editedFirstname, setEditedFirstname] = useState(props.Firstname);
+  const [editedLastname, setEditedLastname] = useState(props.Lastname);
+  const [editedPhone, setEditedPhone] = useState(props.Phone);
+  const [editedAddress, setEditedAddress] = useState(props.Address);
+  const [editedPostalcode, setEditedPostalcode] = useState(props.Postalcode);
+  const [editedCity, setEditedCity] = useState(props.City);
+  const [editedCountry, setEditedCountry] = useState(props.Country);
+  const [editedEmail, setEditedEmail] = useState(props.Email);
+
+  // Save the history so that the editing can be cancelled
+  const [FirstnameHistory, setFirstnameHistory] = useState(props.Firstname);
+  const [LastnameHistory, setLastnameHistory] = useState(props.Lastname);
+  const [PhoneHistory, setPhoneHistory] = useState(props.Phone);
+  const [AddressHistory, setAddressHistory] = useState(props.Address);
+  const [PostalcodeHistory, setPostalcodeHistory] = useState(props.Postalcode);
+  const [CityHistory, setCityHistory] = useState(props.City);
+  const [CountryHistory, setCountryHistory] = useState(props.Country);
+  const [EmailHistory, setEmailHistory] = useState(props.Email);
+
+  const editOnClick = () =>
+  {
+    setEdit(true);
+  }
+
+  const saveEditOnClick = async () =>
+  {
+    // TODO: Send the edited values to the database to actually save the edit
```

Comment:
This comment is unnecessary, right? That's what we're doing, sending the values, no? 🤪